src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -0,0 +1,527 @@
+            isB = (
+                is_heavy_hadron(genlep_pa1G, 5)
+                | is_heavy_hadron(genlep_pa2G, 5)
+                | is_heavy_hadron(genlep_pa3G, 5)
+                | is_heavy_hadron(genlep_pa4G, 5)
+            )
+
+            Genlep = ak.zip(
+                {
+                    "pT": genlep.pt,
+                    "eta": genlep.eta,
+                    "phi": genlep.phi,
+                    "pdgID": genlep.pdgId,
+                    "status": genlep.status,
+                    "mother": ak.fill_none(
+                        ak.where(isB | isD, 5 * isB + 4 * isD, 10 * istau + 100 * isWZ),
+                        0,
+                    ),
+                }
+            )
+
+            ###################
+            #  Gen particles  #
+            ###################
+            # Gen particles of interest in the BTA ttbar workflow
+            genpart = events.GenPart[
+                (events.GenPart.statusFlags & (1 << 7)) > 0
+            ]  # isHardProcess
+            gen_channel = ak.prod(
+                genpart[(abs(genpart.pdgId) == 11) | (abs(genpart.pdgId) == 13)].pdgId,
+                axis=1,
+            )
+
+            gen_arrays = ak.zip(
+                {
+                    "pt": genpart.pt,
+                    "eta": genpart.eta,
+                    "phi": genpart.phi,
+                    "m": genpart.mass,
+                    "id": genpart.pdgId,
+                }
+            )
+
+            #################
+            #  Gen weights  #
+            #################
+            # top pT reweighting, according to BTA code
+            genttbar = events.GenPart[
+                ((events.GenPart.statusFlags & (1 << 13)) > 0)
+                & (abs(events.GenPart.pdgId) == 6)
+            ]  # is last copy and is t/tbar
+            gentop = ak.firsts(genttbar[genttbar.pdgId == 6])
+            genantitop = ak.firsts(genttbar[genttbar.pdgId == -6])
+            basic_vars["ttbar_ptweight"] = ak.fill_none(
+                np.sqrt(
+                    np.exp(0.0615 - 0.0005 * gentop.pt)
+                    * np.exp(0.0615 - 0.0005 * genantitop.pt)
+                ),
+                1.0,
+            )
+
+            # store weights according to BTA code
+            # note: in original BTA code, all weights are appended in a same variable ttbar_w. Here we separate them according to the logic in NanoAOD
+            basic_vars["ttbar_w"] = events.genWeight
+            lhe_pdf_w_arrays = getattr(events, "LHEPdfWeight", None)
+            lhe_scale_w_arrays = getattr(events, "LHEScaleWeight", None)
+            ps_w_arrays = getattr(events, "PSWeight", None)
+
+        ###############
+        #     Jet     #
+        ###############
+        # Jet: same with nominal BTA workflow
+        jet = events.Jet[
+            (events.Jet.pt > 20.0) & (abs(events.Jet.eta) < 2.5)
+        ]  # basic selection
```

Comment:
Could be also added as a basic selection in `selection.py`

Reply:
The jet selection actually includes the ID cut, maybe we add a flag in the selection where jet ID is not implemeted in the `selection.py`

Reply:
This is introduced in Run3 (https://twiki.cern.ch/twiki/bin/view/CMS/JetID13p6TeV) and indeed this is not there in the JetId module in [cmssw](https://github.com/cms-sw/cmssw/blob/CMSSW_13_0_X/PhysicsTools/Heppy/python/physicsobjects/Jet.py#L123-L131
)(not sure I am checking the correct code, @demuller  please correct me if I am wrong)

Reply:
Ok thanks! But looking at the twiki, I wonder why the following is skiped: `Neutral Hadron Fraction < 0.99`, `Neutral EM Fraction< 0.9`

Reply:
this `chEF > 0.01` cut was needed back then to obtain the Run 3 jet ID definition based on the Run 2 UL one, i.e. we don't need any of these additional cuts anymore (also the ones mentioned by @colizz), as they are now contained per default in the jet ID defintion

Reply:
Thanks for the investigation! That's very helpful.

Reply:
For the jet selection in BTA, we can leave this way because the JetID selection is not added, but all stored (as appeared in the original code logic). So analysis can use them for later selection.